### DISCUSSION TITLE
Redirect on sign was working until it stopped. What am I missing?

### DISCUSSION THREAD

**Discussion by j-lee8** [nextauthjs/next-auth#1788] [Help] [answered]
Hi guys.

Apologies for the dumb question! I've just integrated NextAuth with Google OAuth and it signs in and out fine (it redirects correctly on sign out), but the sign in never redirects to /app/main.

It WAS working, but I took the login.tsx page out of page/auth/ and placed it a level higher in pages/ and it stopped redirecting. I adjusted the path accordingly but...it just goes to /login# instead of /app/main. I'm doing something dumb but I've updated the necessary paths so it's mind boggling to me.

Here's the code;
```
// pages/login.tsx     ------ exported from pages/index.tsx
<button onClick={() => signIn('google', { callbackUrl: '/app/main', redirect: true })}>
				Sign In
			</button>
```
```
// [...nextauth].ts
pages: {
		signIn: '/login',
		// signOut: '/login'
  	}
```

```
// pages/index.tsx

export { default } from './login';
```
I've changed the callbackUrl to be absolute and also did a router.push('/app/main') but it just always goes to /login#

**Comment by j-lee8** [comment, marked as answer]
Seems absolute hard coded URLs were the way to solve this. Thanks all.

**Reply by MauriceArikoglu** [reply]
@JavaJamie Hi, could you please give an example for how you solved this? I think I have the same problem but couldnt yet find a solution! Thanks
